### PR TITLE
ci: update codecov workflow to use gcovr

### LINKED ISSUE #66656: codecov reports incomplete
**Describe the bug**
codecov reports are incomplete and missing coverage data. This  can be seen in https://app.codecov.io/github/zephyrproject-rtos/zephyr/blob/main/kernel%2Fmmu.c#L560 for example, where this function is covered but it appears as not covered.

**To Reproduce**
Run coverage locally on some of the kernel tests dealing with memory protection and see that local reports mark k_mem_map as covered, in codecov it is not.

**Expected behavior**
Covered functions should be reported as such.

**Impact**
Confusion about test coverage.

### PR DESCRIPTION
**Pull request by nashif**
- ci: codecov: use gcovr and upload cobertura style file
- kernel: fatal: remove LCOV exclusion

Fixes #66656

### CODE REVIEW

**Comment by kartben** [comment]
> ci: codecov: use gcovr and upload cobertura style file

Could you please make the commit message slightly more explicit? Is the change motivated by addressing a particular bug/issue with lcov? Or just moving to a more modern tool? Something else? 
Thanks!

**Comment by Thalley** [comment]
@nashif Would it make sense to make gcovr the default when we do `twister --coverage`? Seems odd to have CI run the non-default coverage tool

**Review by jhedberg — DISMISSED** [review]

File: .github/workflows/codecov.yaml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name: Code Coverage with codecov
 
 on:
+  workflow_dispatch:
+    branches: [main]
```
Comment (stephanosio) [inline .github/workflows/codecov.yaml:5]: Is this intentional? or a leftover from testing?

Comment (nashif) [reply .github/workflows/codecov.yaml]: removed.

**Comment by nashif** [comment]
> @nashif Would it make sense to make gcovr the default when we do `twister --coverage`? Seems odd to have CI run the non-default coverage tool

yeah, we can do that. @golowanow  can you take care of that?

File: .github/workflows/codecov.yaml (hunk 1 above)
Comment (jhedberg) [inline .github/workflows/codecov.yaml]: It's still here in the first commit. I guess you squashed the removal into the wrong commit?

Comment (nashif) [reply .github/workflows/codecov.yaml]: I did not squash, thats why I mentioned it in the commit message where it was removed.

Comment (nashif) [reply .github/workflows/codecov.yaml]: anyways, squashed now.

**Review by jhedberg — APPROVED** [review]

**Comment by golowanow** [comment]
> > @nashif Would it make sense to make gcovr the default when we do `twister --coverage`? Seems odd to have CI run the non-default coverage tool
> 
> yeah, we can do that. @golowanow can you take care of that?

#66712

**Review by stephanosio — APPROVED** [review]

**Pull request merged**